backend/service/aws/aws.go

```diff
@@ -63,48 +78,90 @@ func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service
-				ScalingLimits: &awsv1.ScalingLimits{
-					MaxReadCapacityUnits:  ds.MaxReadCapacityUnits,
-					MaxWriteCapacityUnits: ds.MaxWriteCapacityUnits,
-					MaxScaleFactor:        ds.MaxScaleFactor,
-					EnableOverride:        ds.EnableOverride,
-				},
-			},
+		c.createRegionalClients(c.currentAccountAlias, region, ac.Regions, ds, regionCfg)
+	}
 
-			s3:          s3.NewFromConfig(regionCfg),
-			kinesis:     kinesis.NewFromConfig(regionCfg),
-			ec2:         ec2.NewFromConfig(regionCfg),
-			autoscaling: autoscaling.NewFromConfig(regionCfg),
-			dynamodb:    dynamodb.NewFromConfig(regionCfg),
-			sts:         sts.NewFromConfig(regionCfg),
-			iam:         iam.NewFromConfig(regionCfg),
-		}
+	if err := c.configureAdditionalAccountClient(ac.AdditionalAccounts, ds, awsHTTPClient, awsClientCommonOptions); err != nil {
+		return nil, err
 	}
 
 	return c, nil
 }
 
+func (c *client) configureAdditionalAccountClient(accounts []*awsv1.AWSAccount, ds *awsv1.ScalingLimits, awsHTTPClient *http.Client, awsClientOptions []func(*config.LoadOptions) error) error {
+	for _, account := range accounts {
+		accountRoleARN := fmt.Sprintf("arn:aws:iam::%s:role/%s", account.AccountNumber, account.IamRole)
+		// For doing STS calls it does not matter which region client we are using, as they are not bounded by region
+		// we choose just the first region client
+		stsClient := c.accounts[c.currentAccountAlias].clients[c.accounts[c.currentAccountAlias].regions[0]].sts
+		assumeRoleProvider := stscreds.NewAssumeRoleProvider(stsClient, accountRoleARN)
+		credsCache := aws.NewCredentialsCache(assumeRoleProvider)
+
+		for _, region := range account.Regions {
+			regionCfg, err := config.LoadDefaultConfig(context.TODO(),
+				append(awsClientOptions, config.WithRegion(region))...,
+			)
+			if err != nil {
+				return err
+			}
+
+			regionCfg.Credentials = credsCache
+
+			c.createRegionalClients(account.Alias, region, account.Regions, ds, regionCfg)
+		}
+	}
+
+	return nil
+}
+
+func (c *client) createRegionalClients(accountAlias, region string, regions []string, ds *awsv1.ScalingLimits, regionCfg aws.Config) {
+	if _, ok := c.accounts[accountAlias]; !ok {
+		c.accounts[accountAlias] = &accountClients{
+			alias:   accountAlias,
+			regions: regions,
+			clients: map[string]*regionalClient{},
+		}
+	}
+
+	c.accounts[accountAlias].clients[region] = &regionalClient{
+		region: region,
+		dynamodbCfg: &awsv1.DynamodbConfig{
+			ScalingLimits: &awsv1.ScalingLimits{
+				MaxReadCapacityUnits:  ds.MaxReadCapacityUnits,
+				MaxWriteCapacityUnits: ds.MaxWriteCapacityUnits,
+				MaxScaleFactor:        ds.MaxScaleFactor,
+				EnableOverride:        ds.EnableOverride,
+			},
+		},
+
+		s3:          s3.NewFromConfig(regionCfg),
+		kinesis:     kinesis.NewFromConfig(regionCfg),
+		ec2:         ec2.NewFromConfig(regionCfg),
+		autoscaling: autoscaling.NewFromConfig(regionCfg),
+		dynamodb:    dynamodb.NewFromConfig(regionCfg),
+		sts:         sts.NewFromConfig(regionCfg),
+	}
```

Comment:
the former regionalClient struct also had the `iam` field. just checking, do you not need it anymore?

Reply:
ooo yeah good catch @scarlettperry! Thank you.

Reply:
It was just added in today https://github.com/lyft/clutch/pull/1863, probably missed it during my rebase.